### PR TITLE
impl FusedFuture on Unordered::next

### LINKED ISSUE #16: `Unordered::next` does not return a `FusedFuture`
`futures::stream::StreamExt::next` returns a future that [explicitly implements `FusedFuture`](https://docs.rs/futures-util/0.3.21/src/futures_util/stream/stream/next.rs.html#22-26). 
`Unordered::next` ought to as well.

This comes up when using `.next()` with the `futures::select!` macro.

### PR DESCRIPTION
**Pull request by daprilik**
Closes #16 

Note: I had to switch `next()` over from an `async` function returning an anonymous future, to a sync function returning an explicit future. This is because leaving it as an `async` function doesn't propagate the new `FusedFuture` impl from the inner `Next` future on to the returned anonymous future.

Alternatively, if we really don't want to leak the `Next` type, we could change the signature to return an `impl Future<Output = Option<<Self as PollNext>::Item>> + FusedFuture + 'a`, but that's a real mouthful...

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -435,21 +435,22 @@ where
     ///     println!("done!");
     /// }
     /// ```
-    pub async fn next(&mut self) -> Option<<Self as PollNext>::Item> {
-        return Next(self).await;
+    pub fn next(&mut self) -> Next<'_, Self> {
```
Comment (eholk) [inline src/lib.rs:438]: I think I have a slight preference for `-> impl Future<Output = Option<<Self as PollNext>::Item>> + FusedFuture + 'a` here, even if it is a bit wordy. In practice, I don't think it matters much since users of the type don't need to type it. It will clutter the docs, I guess.

In general I prefer to reveal as few implementation details as possible since that preserves more semver-compatible options for making changes in the future. 

That said, I would be surprised if this ever changes, and even if it does, once Rust has type-alias-impl-trait, we can just redefine `Next` as `struct Next<'a, T>(impl Future<Output = Option<<Self as PollNext>::Item>> + FusedFuture + 'a)`.

Anyway, I have a slight preference for the `-> impl` version, but I don't think it matters much so I'm fine with either way.

**Review by eholk — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (daprilik) [reply src/lib.rs:438]: Note that another caveat of that approach is that the `FusedFuture` bound would have to be removed when the `futures-rs` feature isn't enabled, and I think the only way of doing that would be to have an entirely different method signature in that case.

Comment (eholk) [reply src/lib.rs:438]: Oh yeah, good point. I'd say that's a point in favor of returning `Next` then.

File: src/lib.rs [hunk 2]
```diff
@@ -1083,4 +1084,13 @@ cfg_futures_rs! {
             streams
         }
     }
+
+    impl<T> FusedFuture for Next<'_, T>
```
Comment (udoprog) [inline src/lib.rs:1088]: Should this impl be conditional on `#[cfg(feature = "futures-rs")]`?

**Review by udoprog — CHANGES_REQUESTED** [review]
One note, otherwise LGTM. Even though it seems like `build_no_default_features` is not doing what it's supposed to!

File: src/lib.rs (hunk 2 above)
Comment (udoprog) [reply src/lib.rs:1088]: Oh, my bad. It is!

**Review by udoprog — APPROVED** [review]

**Comment by udoprog** [comment]
Thank you!

**Comment by udoprog** [comment]
Yeah, of course. Shipped `0.9.2`. Enjoy!
